pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -272,6 +276,20 @@ func (r *ReconcileClusterProvision) reconcileRunningJob(instance *hivev1.Cluster
 
 	pLog = pLog.WithField("job", job.Name)
 
+	installPod, err := r.getInstallPod(job, pLog)
+	if err != nil {
+		pLog.WithError(err).Error("could not get install pod")
+		return r.transitionStage(instance, hivev1.ClusterProvisionStageFailed, "InstallPodMissing", err.Error(), pLog)
+	}
+
+	if installPod.Status.Phase == "Pending" {
+		pLog.WithField("pod", installPod.Name).Error("install pod is stuck")
+		if err := r.setCondition(instance, hivev1.ClusterProvisionInitializedCondition, corev1.ConditionFalse, "InstallPodStuck", "install pod is stuck in pending phase", controllerutils.UpdateConditionAlways, pLog); err != nil {
```

Comment:
I think that I would rather have a separate PodStuck condition that can be set True rather than setting the Initialized condition to False.